pkg/api/validation/validation.go

```diff
@@ -30,12 +30,29 @@ import (
 	"github.com/golang/glog"
 )
 
+const MinPort int = 1
+const MinHostPort int = 0
```

Comment:
IIUC 0 is allowed as it's default value if HostPort in not explicitly specified. I can change this but I'm not sure if we want to.